include/Keymap.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Keymap.h - holds information about a key mapping
+ *
+ * Copyright (c) 2020 Martin Pavelek <he29.HS/at/gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef KEYMAP_H
+#define KEYMAP_H
+
+#include <vector>
+#include <QObject>
+#include <QString>
+
+#include "Note.h"
+#include "SerializingObject.h"
+
+class Keymap : public QObject, public SerializingObject
```

Comment:
I'm not quite sure. When I started developing stuff for LMMS a few years ago, I often ran into problems because I did _not_ inherit from the `QObject` or did not add the `Q_OBJECT` macro. So I started to adding it by default. :)

...and, trying to build it now without `QObject`, I can see it fails on undefined `tr()`. So apparently translations for the default _empty_ scales and keymaps would not work. It's just a line in the constructor, though, so if it is a problem, I could replace it with something language-neutral, like `---`.

Reply:
I think this is fine to leave for now then. We're going to have to solve this problem more generally as we work to reduce the use of Qt in the core.